docs/reference/flow/01-diagrams.md

```diff
@@ -24,29 +24,92 @@ flowchart LR
     storage --> st
 ```
 
-## Token Balance Actions
+## Token Flows
 
 ```mermaid
-flowchart LR
-    erc_transfers[(ERC20 Transfer Actions)]
-    dep([Deposit - add])
-    ref([Refund - remove])
-    wtd([Withdraw - remove])
-
-    erc_transfers --> dep
-    erc_transfers --> ref
-    erc_transfers --> wtd
+sequenceDiagram
+  actor Sender
+
+  Sender ->> Flow: deposit()
+  activate Flow
+  Sender -->> Flow: Transfer tokens
+  deactivate Flow
+
+  Sender ->> Flow: refund()
+  activate Flow
+  Flow -->> Sender: Transfer tokens
+  deactivate Flow
+
+  Sender ->> Flow: withdraw()
+  activate Flow
+  Create actor Recipient
+  Flow -->> Recipient: Transfer tokens
+  deactivate Flow
```

Comment:
from this diagram is not clear if only the sender can call `withdraw`. let's add an arrow from the recipient to the flow contract as well

Reply:
This diagram is about token flow based on function call and not about access control. Adding arrows from recipient will complicate it. What you are suggesting it included in "Flow Actors with Actions" diagram.

Reply:
How about calling it User instead of Sender in this diagram?

I have a better idea, I can split it into sub categories.

<img width="200" alt="Screenshot 2024-12-05 at 11 02 14" src="https://github.com/user-attachments/assets/f433a11f-fd9d-43b4-8bb0-92ae5ba55e96">

Reply:
hmm, in case of deposit/withdraw it would work, but given that `refund` is callable only by sender, not sure

Reply:
Like this.

<img width="300" alt="Screenshot 2024-12-05 at 11 09 12" src="https://github.com/user-attachments/assets/6d9cc2d0-7fd5-4804-b241-13889eacda66">
<img width="300" alt="Screenshot 2024-12-05 at 11 09 15" src="https://github.com/user-attachments/assets/f9b8bdc8-6692-4182-a4b1-e68c450d67fd">
<img width="300" alt="Screenshot 2024-12-05 at 11 09 18" src="https://github.com/user-attachments/assets/7d24e8af-88ed-41a1-8270-20d50237d0fa">

Reply:
ok, agree, let's use "User"